lib/utils.pm

```diff
@@ -2044,4 +2045,26 @@ sub permit_root_ssh_in_sol {
     enter_cmd("[ `grep \"^PermitRootLogin *yes\" $sshd_config_file | wc -l` -gt 0 ] || (echo 'PermitRootLogin yes' >>$sshd_config_file; systemctl restart sshd)");
 }
 
+=head2 cleanup_disk_space
+    cleanup_disk_space();
+
+In fully_patch_system and minimal_patch_system, we'll create so many
+snapshots. which will cost large part of the disk space. We need to
+delete these snapshots before the migration. We need to remove snapper-zypp-plugin
+to avoid creating new snapshots from 'zypper install'. see bsc#1192331.
+
+=cut
+
+sub cleanup_disk_space {
+    record_soft_failure "bsc#1192331", "Low diskspace on Filesystem root";
+
+    my @snap_lists = split /\n/, script_output("snapper list --disable-used-space | grep important= | grep -v single | awk \'{print \$1}\'");
+    foreach my $snapid (@snap_lists) {
+        assert_script_run("snapper delete -s $snapid", timeout => 120) if ($snapid > 3);
+    }
+
+    # remove snapper-zypp-plugin if it was installed.
+    assert_script_run('zypper remove -y snapper-zypp-plugin ||:');
```

Comment:
@hjluo 
I am not an expert on snapshot, however, can you please consider setting max number of the snapshots rather than removing the package?

It will disable snapshot function, correct?

Reply:
NO, To disable installation snapshots, uninstall the package snapper-zypp-plugin.

Reply:
Got it, thanks.

Anyway, removing an installed package is not a better solution here. can you try to enhance this?